pynest/examples/Potjans_2014/network.py

```diff
@@ -0,0 +1,506 @@
+
+    Arguments
+    ---------
+    sim_dict
+        dictionary containing all parameters specific to the simulation
+        such as the directory the data is stored in and the seeds
+        (see: sim_params.py)
+    net_dict
+         dictionary containing all parameters specific to the neurons
+         and the network (see: network_params.py)
+
+    Keyword Arguments
+    -----------------
+    stim_dict
+        dictionary containing all parameter specific to the stimulus
+        (see: stimulus_params.py)
+
+    """
+    def __init__(self, sim_dict, net_dict, stim_dict=None):
+        self.sim_dict = sim_dict
+        self.net_dict = net_dict
+        if stim_dict is not None:
+            self.stim_dict = stim_dict
+        else:
+            self.stim_dict = None
+        self.data_path = sim_dict['data_path']
+        if nest.Rank() == 0:
+            if os.path.isdir(self.sim_dict['data_path']):
+                print('data directory already exists')
+            else:
+                os.mkdir(self.sim_dict['data_path'])
+                print('data directory created')
+            print('Data will be written to %s' % self.data_path)
+
+    def setup_nest(self):
+        """ Hands parameters to the NEST-kernel.
+
+        This function resets the NEST-kernel and passes parameters to it.
+        The number of seeds for the NEST-kernel is computed, based on the
+        total number of MPI processes and threads of each.
+
+        Parameters
+        """
+        nest.ResetKernel()
+        master_seed = self.sim_dict['master_seed']
+        if nest.Rank() == 0:
+            print('Master seed: %i ' % master_seed)
+        nest.SetKernelStatus(
+            {'local_num_threads': self.sim_dict['local_num_threads']}
+            )
+        N_tp = nest.GetKernelStatus(['total_num_virtual_procs'])[0]
+        if nest.Rank() == 0:
+            print('Number of total processes: %i' % N_tp)
+        rng_seeds = list(
+            range(
+                master_seed + 1 + N_tp,
+                master_seed + 1 + (2 * N_tp)
+                )
+            )
+        grng_seed = master_seed + N_tp
+        if nest.Rank() == 0:
+            print(
+                'Seeds for random number generators of virtual processes: %r'
+                % rng_seeds
+                )
+            print('Global random number generator seed: %i' % grng_seed)
+        self.pyrngs = [np.random.RandomState(s) for s in list(range(
+            master_seed, master_seed + N_tp))]
+        self.sim_resolution = self.sim_dict['sim_resolution']
+        kernel_dict = {
+            'resolution': self.sim_resolution,
+            'grng_seed': grng_seed,
+            'rng_seeds': rng_seeds,
+            'overwrite_files': self.sim_dict['overwrite_files'],
+            'print_time': self.sim_dict['print_time'],
+            }
+        nest.SetKernelStatus(kernel_dict)
+
+    def create_populations(self):
+        """ This function creates the neuronal populations.
```

Comment:
see above, please rephrase to "Creates..."
please repeat this for all other functions.